packages/grid/_modules_/grid/constants/eventsConstants.ts

```diff
@@ -1,85 +1,431 @@
 // Web standard events
+
+/**
+ * Fired when the grid is resized. Called with a [[GridResizeParams]] object.
+ * @event
+ */
 export const GRID_RESIZE = 'resize';
+
+/**
+ * Fired when a `focusout` event happens in the grid.
+ * @event
+ */
 export const GRID_FOCUS_OUT = 'focusout';
+
+/**
+ * Fired when a `keydown` event happens in the grid.
+ * @event
+ */
 export const GRID_KEYDOWN = 'keydown';
+
+/**
+ * Fired when a `keyup` event happens in the grid.
+ * @event
+ */
 export const GRID_KEYUP = 'keyup';
+
+/**
+ * @ignore - do not document.
+ */
 export const GRID_NATIVE_SCROLL = 'scroll';
-export const GRID_DRAGEND = 'dragend';
 
 // GRID events
+
+/**
+ * Fired when an exception is thrown in the grid.
+ * @event
+ */
 export const GRID_COMPONENT_ERROR = 'componentError';
+
+/**
+ * Fired when the grid is unmounted.
+ * @event
+ */
 export const GRID_UNMOUNT = 'unmount';
+
+/**
+ * @ignore - do not document.
+ */
 export const GRID_ELEMENT_FOCUS_OUT = 'gridFocusOut';
 
+/**
+ * Fired when the mode of a cell changes. Called with a [[GridCellModeChangeParams]] object.
+ * @event
+ */
 export const GRID_CELL_MODE_CHANGE = 'cellModeChange';
+
+/**
+ * Fired when a cell is clicked. Called with a [[GridCellParams]] object.
+ * @event
+ */
 export const GRID_CELL_CLICK = 'cellClick';
+
+/**
+ * Fired when a cell is double-clicked. Called with a [[GridCellParams]] object.
+ * @event
+ */
 export const GRID_CELL_DOUBLE_CLICK = 'cellDoubleClick';
+
+/**
+ * Fired when a `mousedown` event happens in a cell. Called with a [[GridCellParams]] object.
+ * @event
+ */
 export const GRID_CELL_MOUSE_DOWN = 'cellMouseDown';
```

Comment:
Should we have this level of granularity when exposing events? I think we should document only the most obvious events (click, double-click, keydown) and wait users to request more.

Reply:
IMO it is part of the api so ppl should be free to use it. I don't see the issue.

Documenting only the most obvious is not a proper api documentation as there will be missing things so the doc will be incomplete as soon as it's updated. Ppl will have to look at the typings to get the most accurate doc.

Reply:
The issue is about the long tail of use cases. What about touch start, touch move, touch end, or even the pointers events https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#event_types_and_global_event_handlers?

Reply:
I think we should think about that users would potentially create with these events. We are exposing all the column header drag events, but I don't think users would write another drag-and-drop feature from scratch. Having all the event we use internally makes the documentation long and may force users to question themselves: which event should I subscribe? `mousedown` or `click`?

Reply:
or force us to hold on breaking changes until the next breaking changes release window.

Reply:
I marked more events with @ignore. My rationale is:
- Ignore drag-and-drop events because they have a very specific use. 
- Ignore the less obvious events published by column headers (keep only click, double and keydown).
- Ignore global events (`GRID_FOCUS_OUT`) because users can already subscribe to them using the `ref` prop. Also, if we're going to document them we should document EVERYTHING.

Reply:
This is from ag grid https://www.ag-grid.com/react-grid/grid-events/

rowDataChanged | The client has set new data into the grid using api.setRowData() or by changing the rowData bound property.
-- | --
rowDataUpdated | The client has updated data for the grid using api.applyTransaction(transaction) or by changing the rowData bound property with immutableData=true.


rowDragEnter | A drag has started, or dragging was already started and the mouse has re-entered the grid having previously left the grid.
-- | --
rowDragMove | The mouse has moved while dragging.
rowDragLeave | The mouse has left the grid while dragging.
rowDragEnd | The drag has finished over the grid.



RowEvent 
            ├── RowSelectedEvent {}
            ├── RowClickedEvent {}
            ├── RowDoubleClickedEvent {}
            ├── RowEditingStartedEvent {}
            ├── RowEditingStoppedEvent {}
            ├── RowGroupOpenedEvent {
            ├── RowValueChangedEvent {}
            ├── VirtualRowRemovedEvent {}
            └── CellEvent 
                ├── CellClickedEvent {}
                ├── CellMouseDownEvent {}
                ├── CellDoubleClickedEvent {}
                ├── CellMouseOverEvent {}
                ├── CellMouseOutEvent {}
                ├── CellContextMenuEvent {}
                ├── CellEditingStartedEvent {}
                ├── CellKeyDown {}
                ├── CellKeyPress {}
                ├── CellEditingStoppedEvent 
                └── CellValueChangedEvent